### ISSUE TITLE
Middle clicking scrollbar causes otter to enter scroll mode

### ISSUE THREAD

**Issue by cofi89** [OtterBrowser/otter-browser#1006]
After middle clicking on scrollbar and moving cursor back in page, moving it causes otter to scroll the page, as in scroll mode ( middle click in page ).  
However no "scroll mode" cursor is shown, it stays normal.  
Left or right click stops it.  

What's interesting, is that I can still enter and exit "normal"  scroll mode as usual with middle click, but exiting it doesn't kill the "scrollbar middle click mode". It just keeps scrolling with mouse movement regardless, until killed with left or right click.

Clean cache, profile, and clean (non-optimized) build make no difference. 
No output from terminal.

IIRC it started at about f5af35283

Sys info:  
* Otter = latest git build
* Debian Jessie
* QT 5.3

**Comment by Emdek** [comment]
@cofi89, it appears to be some unwanted feature that used to be blocked before.
I have a patch that takes care of it but I need to test it a bit first.

**Comment by cofi89** [comment]
It fixed the "scroll mode" allright, however middle click on scrollbar no longer works.

**Comment by cofi89** [comment]
@Emdek, this is still an issue ( for me atleast ).

After some poking around the code in "QtWebKitWebWidget.cpp" & reverting related commits, I've found that removing the following solves the problem:
```
else if (event->type() == QEvent::MouseButtonRelease && mouseEvent && mouseEvent->button() == Qt::MiddleButton && !getCurrentHitTestResult().flags.testFlag(IsContentEditableTest))
{
	return true;
}
```
https://github.com/OtterBrowser/otter-browser/blob/master/src/modules/backends/web/qtwebkit/QtWebKitWebWidget.cpp#L2394-L2397

It was introduced in: 99dc83cd  

----------

Now, ignoring that I have almost no knowledge of c++/qt, after some more poking & transplanting I got a fix ( hopefully one that's usable ).  
Ignoring scrollbar middle click [in the line I've mentioned](https://github.com/OtterBrowser/otter-browser/blob/master/src/modules/backends/web/qtwebkit/QtWebKitWebWidget.cpp#L2394) fixes this issue, and does't affect #932 ( as far as I can test ):
```
else if (event->type() == QEvent::MouseButtonRelease && mouseEvent && mouseEvent->button() == Qt::MiddleButton && !getCurrentHitTestResult().flags.testFlag(IsContentEditableTest))
```
changed to:
```
else if ((event->type() == QEvent::MouseButtonRelease && mouseEvent && mouseEvent->button() == Qt::MiddleButton && !getCurrentHitTestResult().flags.testFlag(IsContentEditableTest)) && !isScrollBar(mouseEvent->pos()))
```
I obviously have no idea what, if any, side effects that might have elsewhere, but if it's any good and of any use, then I'm glad to have helped. :smiley:
